### PR TITLE
Firestore service utils

### LINKED ISSUE #32: Firestore Service Utilities
## Purpose
Several interactions need to be made with the Firestore database at various stages in the application. Having individual implementations of data transactions is not conducive for an easily testable or maintainable code base. Developing a single file where either a service class or utility functions handle the data communication supports DRY (Don't-Repeat-Yourself) and testable code.
## Acceptance Criteria
- Current and relevant data structures have an associated model (User, Pet, Feeder)
- For each model, there is a utility that reads data from the database and parses it according to the model
- For each model, there is a utility that packages data according to the data organization in the database and writes the data

### PR DESCRIPTION
**Pull request by GuassianFlux**
Closes #32 

Everything should be there to unblock the other issues. Let me know if there is anything I should address before the merge.

### CODE REVIEW

File: lib/model/Pet.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  Map<String, dynamic> toJson() {
+    return {
+      'RFID': this.rfid,
+      'dob': this.dob,
+      'id': this.petID,
+      'name': this.name,
+      'type': this.type,
+      'lbs': this.lbs,
+    };
+  }
```
Comment (jlp0063) [inline lib/model/Pet.dart]: Here I noticed you changed from using petID to id, between your getters and setters. Can we just use petID for both getter and setter to make it more readable. At first glance, I might think that id is a separate field from petID.

File: lib/model/Schedule.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        time: parsedJson['time'] ?? '');
+  }
+
+  Map<String, dynamic> toJson() {
+    return {
+      'id': this.schedID,
+      'portion': this.portion,
+      'time': this.time,
+    };
+  }
```
Comment (jlp0063) [inline lib/model/Schedule.dart]: Same as comment from Pet.dart

File: lib/model/Feeder.dart [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        name: parsedJson['name'] ?? '');
+  }
+
+  Map<String, dynamic> toJson() {
+    return {
+      'id': this.feederID,
+      'modelNumber': this.modelNum,
+      'name': this.name
+    };
+  }
```
Comment (jlp0063) [inline lib/model/Feeder.dart]: Same as comment from pet.dart

**Review by jlp0063 — DISMISSED** [review]

File: lib/model/Pet.dart (hunk 1 above)
Comment (wsciaroni) [reply lib/model/Pet.dart]: I definitely agree that we should use more readable names in the json fields.

So for this I think @jlp0063 is saying we should have the json have the following fields:

- `rfid`
- `dob`
- `petID`
- `name`
- `type`
- `lbs`

In this way they map perfectly 1:1 with the fields in the actual Pet structure.

**Review by wsciaroni — APPROVED** [review]
Looks good after those changes!

**Review by jalobrano — APPROVED** [review]

**Review by jlp0063 — APPROVED** [review]

**Pull request merged**